dsp-value-objects/src/main/scala/dsp/valueobjects/V2.scala

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.valueobjects
+
+import java.util.UUID
+import java.util.Base64
+import java.nio.ByteBuffer
+import scala.util.matching.Regex
+import org.apache.commons.validator.routines.UrlValidator
+import org.apache.commons.lang3.StringUtils
+import com.google.gwt.safehtml.shared.UriUtils.encodeAllowEscapes
+
+// TODO-mpro: don't forget to remove all occurances and additional "helper"
+// implementations in webapi project which needed to be added temporary in order
+// to avoid circular dependencies after moving value objects to separate project.
+object V2 {
+  val DE: String = "de"
+  val EN: String = "en"
+  val FR: String = "fr"
+  val IT: String = "it"
+  val RM: String = "rm"
+
+  val SupportedLanguageCodes: Set[String] = Set(
+    DE,
+    EN,
+    FR,
+    IT,
+    RM
+  )
+
+  /**
+   * Represents a string with language iso. Allows sorting inside collections by value.
+   *
+   * @param value    the string value.
+   * @param language the language iso.
+   */
+  case class StringLiteralV2(value: String, language: Option[String])
+
+  /**
+   * An exception indicating that the request parameters did not make sense.
+   *
+   * @param message a description of the error.
+   */
+  case class BadRequestException(message: String, cause: Throwable = null) extends Exception(message, cause)
```

Comment:
as this has proven a bit ugly (needed you to touch `KnoraExceptionHandler`), now I'm wondering: Could the instances where this is used just return the error message string (with a Validation, or so), and then in the old code base it gets translated into an actual error? Or would that be a too big refactor?